src/components/common.js

```diff
@@ -555,3 +556,43 @@ export const ClipboardButton = ({ text, onClick, children, ...props }) => {
 export const HeaderRenderer = ({ name, label, sort, onSort, style, ...props }) => h(MiniSortable, { sort, field: name, onSort }, [
   div({ style: { fontWeight: 600, ...style }, ...props }, [label || Utils.normalizeLabel(name)])
 ])
+
+export const DeleteConfirmationModal = ({ title, children, confirmationPrompt = 'Delete', buttonText = 'Delete', onConfirm, onDismiss }) => {
```

Comment:
I wanted a reusable component for the pattern currently used for confirming deleting columns and workspaces. I figured I'd keep this PR minimal and change the existing delete modal later. But as you pointed out, since there's a few inconsistencies, better to go ahead and change it here.

Reply:
What do you think about calling the dialog "ConfirmationModal"? I'm afraid with the current name people won't realize it can be customized for other purposes.

Reply:
PromptedConfirmationDialog?

Reply:
Sounds good to me.